releasenotes/notes/add-coupling-map-to-hls-f8d377ca98f5e1fc.yaml

```diff
@@ -0,0 +1,50 @@
+---
+features:
+  - |
+    Added the ability to pass the coupling map to :class:`.HighLevelSynthesis`
+    transpiler pass. When the coupling map is not ``None``, each synthesis method
+    (built on top of :class:`.HighLevelSynthesisPlugin`) is required to synthesize
+    a given high-level object adhering to this coupling map, or return ``None`` if
+    unable to do so.
+  - |
+    Added two high-level-synthesis plugins for :class:`.LinearFunction`. The
+    :class:`.PMHSynthesisLinearFunction` is the synthesis plugin built on top
+    of the Patel-Markov-Hayes method. As the underlying method does not take the
+    coupling map into account, the plugin only synthesizes the given linear function
+    when the coupling map is ``None``, and returns ``None`` otherwise. The
+    :class:`.KMSSynthesisLinearFunction` is the synthesis plugin built on top of the
+    Kutin-Moulton-Smithline method. The underlying synthesis method synthesizes with
+    respect to linear nearest-neighbour (LNN) architecture. The plugin synthesizes
+    the given linear function either when the coupling map is ``None``, or when there
+    exists a hamiltonian path through the qubits over which this linear function is
+    defined. When the coupling map is not ``None`` and the hamiltonian path does
+    not exist, this plugin returns ``None``.
+upgrade:
```

Comment:
This second note is really a feature note too, but as I mentioned above I feel like this needs to be a separate PR.

```suggestion
```